controllers/storagerequest/storagerequest_controller.go

```diff
@@ -345,6 +353,25 @@ func (r *StorageRequestReconciler) reconcilePhases() (reconcile.Result, error) {
 			r.StorageRequest.Status.Phase = v1alpha1.StorageRequestReady
 		}
 
+		clusterID := util.GetClusterID(r.ctx, r.Client, &r.log)
+		if clusterID != "" {
+			if strings.HasSuffix(r.storageConsumer.Name, clusterID) {
+				updateConsumer = util.AddAnnotation(
+					r.storageConsumer,
+					defaults.StorageConsumerTypeAnnotation,
+					defaults.StorageConsumerTypeLocal,
+				) || updateConsumer
+			}
+		} else {
+			return reconcile.Result{}, fmt.Errorf("failed to get cluster id")
+		}
```

Comment:
instead of if else, fail early if the clusterID is empty?

Reply:
done.